rust/src/lib/nm/profile.rs

```diff
@@ -74,6 +91,43 @@ pub(crate) fn perpare_nm_conns(
             nm_acs,
             gen_conf_mode,
         )? {
+            let for_apply_fwd = iface
+                .base_iface()
+                .ipv4
+                .as_ref()
+                .and_then(|ip4| ip4.forwarding);
+            let current_fwd = exist_nm_conns
+                .iter()
+                .find(|c| c.uuid() == nm_conn.uuid())
+                .and_then(|c| c.ipv4.as_ref())
+                .and_then(|ip4| {
+                    ip4.forwarding.and_then(|val| match val {
+                        0 => Some(false),
+                        1 => Some(true),
+                        _ => None,
+                    })
+                });
+
+            // Remove ipv4.forwarding if interface did NOT explicitly request it,
+            // or if the desired setting matches the current (i.e., no change needed)
```

Comment:
Why do we need to do this? It is not common in nmstate at all, IMO. In an nmstate managed interface we can just force a `forwarding` value to the NM connection. If it's specified in the desired state, use it; if not, take it from the current state.

Or I am missing something? @cathay4t what do you think?

Reply:
```rust
nm_setting.forwarding = iface_ip.forwarding.map(i32::from)
```

Single line is enough, nmstate already has code handling the merging stuff.

Reply:
Since nmstate already handles merging stuff, it is possible that nmstate will apply the forwarding that is actually retrieved by nispor. The intention of having the code here is to drop the forwarding setting when it is not explicitly desired by the user, or the forwarding for apply is equal to the current forwarding in the connection, so that we are not risking applying the forwarding in a system of which the NM does not support forwarding yet. The idea is similar to the commit f38da12c4 (mptcp: Fix mptcp error on older NetworkManager).